src/blueteam/Archon.java

```diff
@@ -2,34 +2,40 @@
 
 import battlecode.common.Direction;
 import battlecode.common.GameActionException;
-import battlecode.common.MapLocation;
 import battlecode.common.RobotController;
+import battlecode.common.RobotInfo;
 
 public class Archon extends Robot {
 
+	private final int desiredNumberOfGardeners = 10;
+
 	Archon(RobotController rc) {
 		super(rc);
 	}
 
 	@Override
 	void step() throws GameActionException {
-		// Generate a random direction
-		Direction dir = randomDirection();
+		// move away from enemy
+		RobotInfo[] robots = rc.senseNearbyRobots(-1, enemy);
+		if (robots.length > 0) {
+			Direction toEnemy = rc.getLocation().directionTo(robots[0].getLocation());
+			tryMove(toEnemy.opposite());
+		} else {
+			tryMove(randomDirection());
+		}
 
-		// Randomly attempt to build a gardener in this direction
-		if (rc.canHireGardener(dir) && Math.random() < .01) {
+		// randomly attempt to build a gardener if we need more
+		Direction dir = randomDirection();
+		int numOfGardeners = rc.readBroadcast(0);
+		if (numOfGardeners < desiredNumberOfGardeners && rc.canHireGardener(dir) && Math.random() < 0.5) {
 			rc.hireGardener(dir);
+			rc.broadcast(0, numOfGardeners + 1);
 		}
 
-		// Move randomly
-		tryMove(randomDirection());
-
-		// Broadcast archon's location for other robots on the team to
-		// know
-		MapLocation myLocation = rc.getLocation();
-		rc.broadcast(0, (int) myLocation.x);
-		rc.broadcast(1, (int) myLocation.y);
-
+		// buy some victory points randomly
+		if (Math.random() < 0.05) {
+			rc.donate(rc.getTeamBullets() / 4);
```

Comment:
Of course it should. 
There still is the problem that if we have multiple archons they drain the bullets so fast that we never can buy a soldier. This might be fixed automatically once we'll be obtaining bullets faster, but I think it should be fixed anyway. Perhaps divide the probability by the number of archons?